sale_line_service_qty_delivered/models/sale_order_line.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+from odoo.tools import float_is_zero, groupby
+
+QTY_DELIVERED_METHOD_DELIVERY = "delivery"
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    qty_delivered_method = fields.Selection(
+        selection_add=[(QTY_DELIVERED_METHOD_DELIVERY, "Delivery")]
+    )
+
+    def _is_delivered_method_delivery(self):
+        self.ensure_one()
+        return (
+            self.product_id.type == "service"
+            and self.product_id.invoice_policy == QTY_DELIVERED_METHOD_DELIVERY
```

Comment:
It's different fields with different selections, don't use the same variable. Just put the value here
```suggestion
            and self.product_id.invoice_policy == "delivery"
```